django/db/backends/postgresql/introspection.py

```diff
@@ -154,64 +154,38 @@ def get_constraints(self, cursor, table_name):
         """
         constraints = {}
         # Loop over the key table, collecting things as constraints
-        # This will get PKs, FKs, and uniques, but not CHECK
         cursor.execute("""
             SELECT
-                kc.constraint_name,
-                kc.column_name,
-                c.constraint_type,
-                array(SELECT table_name::text || '.' || column_name::text
-                      FROM information_schema.constraint_column_usage
-                      WHERE constraint_name = kc.constraint_name)
-            FROM information_schema.key_column_usage AS kc
-            JOIN information_schema.table_constraints AS c ON
-                kc.table_schema = c.table_schema AND
-                kc.table_name = c.table_name AND
-                kc.constraint_name = c.constraint_name
+                c.conname,
+                array(SELECT attname
+                    FROM pg_attribute AS ca
+                    WHERE ca.attrelid = c.conrelid AND
+                        ca.attnum = ANY(c.conkey)),
+                c.contype,
+                (SELECT fkc.relname || '.' || fka.attname
+                    FROM pg_attribute AS fka
+                    JOIN pg_class AS fkc ON
+                        fka.attrelid = fkc.oid
+                    WHERE fka.attrelid = c.confrelid AND
+                        fka.attnum = c.confkey[1])
+                FROM pg_constraint AS c
+                JOIN pg_class AS cl ON
```

Comment:
It would be nice to align the various `FROM`s and `JOIN`s with there corresponding `SELECT`s (like in all other queries of the file) for better readability.
Also, all other queries seem to prefer putting `AND`s at the beginning of the line.